syntax = "proto3";
package org.feuyeux.grpc;

option java_multiple_files = true;
option java_package = "org.feuyeux.grpc.proto";
option java_outer_classname = "Landing";
option go_package = "common/pb";

service LandingService {
  //Unary RPC
  rpc talk (TalkRequest) returns (TalkResponse) {
  }
  //Server streaming RPC
  rpc talkOneAnswerMore (TalkRequest) returns (stream TalkResponse) {
  }
  //Client streaming RPC
  rpc talkMoreAnswerOne (stream TalkRequest) returns (TalkResponse) {
  }
  //Bidirectional streaming RPC
  rpc talkBidirectional (stream TalkRequest) returns (stream TalkResponse) {
  }
}

message TalkRequest {
  //query information
  string data = 1;
  //query meta-info
  string meta = 2;
}

message TalkResponse {
  int32 status = 1;
  repeated TalkResult results = 2;
}

message TalkResult {
  int64 id = 1;
  ResultType type = 2;
  map<string, string> kv = 3;
}

enum ResultType {
  OK = 0;
  FAIL = 1;
}